### ISSUE TITLE
Windows Bcrypt AEAD API handling of nonce

### ISSUE THREAD

**Issue by huitema** [h2o/picotls#271]
I am trying to write a crypto provider for Windows that would parallel "openssl" or "minicrypto", so that picotls could be used on Windows without requiring a port of openssl. I am having a problem with the initialization of AEAD encryption or decryption.

Picotls initializes AEAD using:
~~~
inline void ptls_aead_encrypt_init(ptls_aead_context_t *ctx, uint64_t seq, const void *aad, size_t aadlen)
{
    uint8_t iv[PTLS_MAX_IV_SIZE];

    ptls_aead__build_iv(ctx, iv, seq);
    ctx->do_encrypt_init(ctx, iv, aad, aadlen);
}
~~~
But Windows Bcrypt performs the same function by initializing the structure
`BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO` which contains as parameters, `nonce`, `iv` and `aad`. I have to inialize that structure when implementing the Bcrypt version of `do_encrypt_init()`, but I am not getting access to the `nonce` parameter -- it is already combined into the `iv` string by the call to  `ptls_aead__build_iv`.

Any suggestion how to handle that?

**Comment by kazuho** [comment]
Would it be possible to always do the following?
* set `nonce` of `BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO` to always zero
* set `iv` to the iv generated by `ptls_aead__build_iv`

The assumption here is that the Bcrypt function XORs nonce and iv in particular way (likely as defined by RFC 5116). If that is the case, setting either to zero would solve the issue.

**Comment by huitema** [comment]
Yes, that's what I was planning to do. I think it will work for AES-GCM. We can always look at different solutions when considering other algorithms.

**Comment by huitema** [comment]
Actually, I was misreading the bcrypt API. The input to the API is called a "nonce", but that nonce is exactly what the PTLS API calls IV.

Fixed in PR #275
